### DISCUSSION TITLE
Adguard-home /var/log/messages i/o timeout

### DISCUSSION THREAD

**Discussion by haldenyang** [AdguardTeam/AdGuardHome#7164] [Q&A]
Platform (OS and CPU architecture)

Linux, AMD64 centos7

Installation

snap/manual install

Setup

On one machine

AdGuard Home version

v0.107.52

1.snap logs adguard-home has large messages (also /var/log/messages):

Aug  4 03:45:42 DNS adguard-home.adguard-home: 2024/08/04 03:45:42.307750 [error] dnsproxy: xxx.xxx.xxx.xxx:53: response received over udp: "exchanging with xxx.xxx.xxx.xxx:53 over udp: read udp 192.168.x.x:41524->xxx.xxx.xxx.xxx:53: i/o timeout"
Aug  4 03:45:42 DNS adguard-home.adguard-home: 2024/08/04 03:45:42.310067 [error] dnsproxy: upstream xxx.xxx.xxx.xxx:53 failed to exchange ;1608786975.D4EE072E457C.device.hiwifi.com.#011IN#011 A in 20.004190253s: exchanging with xxx.xxx.xxx.xxx:53 over udp: read udp 192.168.x.x:41524->xxx.xxx.xxx.xxx:53: i/o timeout

2.Average processing time too high,6300ms
3.Average upstream response time high,2000-5000ms

what can I do ?I want to reslove  the problem，thanks ！

**Comment by rodpgh** [comment]
I am having the same issue, let me know if you found a fix

**Comment by yanxu4780** [comment]
same issue, I think is the system problem, Centos 7 too old not woking

**Comment by solidslush** [comment]
I was having similar issues and noticed that my Raspberry Pi5 could also not resolve DNS.  I don't know if it's a problem with the Pi5 or if my upstream limits me after a time (9.9.9.9) in my case.  I changed my DNS to be 9.9.9.9, 149,112,112,112, 1.1.1.1, 1.0.0.1.  So hopefully between Quad9 & Cloudflare something responds.  I suppose I could set my Adguard upstreams to just Quad9 and my Pi to just Cloudflare to see if narrows down where the block is.

I could see being limited, I have a couple of containers that a really chatty and I'm getting 50k requests daily on the same url; one of the reasons I put in Adguard was to answer that request locally.  Though I changed most of those containers to use the IP instead of looking it up.  Maybe that'll bring the chatter down a bit.
